internal/client/dast_external_scan_create.go

```diff
@@ -0,0 +1,86 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/nullify-platform/cli/internal/models"
+	"github.com/nullify-platform/logger/pkg/logger"
+)
+
+type DASTCreateExternalScanInput struct {
+	AppName string `json:"appName"`
+
+	Progress  *int       `json:"progress"`
+	Status    *string    `json:"status"`
+	StartTime *time.Time `json:"startTime"`
+	EndTime   *time.Time `json:"endTime"`
+
+	models.RequestProvider
+	models.RequestDashboardTarget
+}
+
+type DASTCreateExternalScanOutput struct {
+	ScanID string `json:"scanId"`
+}
+
+func (c *NullifyClient) DASTCreateExternalScan(githubOwner string, input *DASTCreateExternalScanInput) (*DASTCreateExternalScanOutput, error) {
+	requestBody, err := json.Marshal(input)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info(
+		"creating external scan",
+		logger.String("appName", input.AppName),
+		logger.String("baseURL", c.BaseURL),
+	)
+
+	req, err := http.NewRequest(
+		"POST",
+		fmt.Sprintf("%s/dast/external?githubOwner=%s", c.BaseURL, githubOwner),
+		strings.NewReader(string(requestBody)),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Below code duplicates code from dast_external_scan_update.go should be better to make this both one function and call it here.

Reply:
that can be done later, maximising code reuse isnt a concern right now